### ISSUE TITLE
[Bug]: Coloured frames in Info-Weapons dialogue window

### ISSUE THREAD

**Issue by BM-W1** [naev/naev#2415]
### What happened?

e.g. the Photo-Voltaic Nanobot Coating has got a frame around the box, that means it could be activated. Isn't it always switched on? The same issue happens with various components such as the place holdes for guardian overseer systems.

### Version

0.11.0 alpha (current)

### What operating system are you seeing the problem on?

Linux

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bobbens** [comment]
Is this when you click on it? I've tested it locally and haven't seen anything out of the ordinary.

Weapon sets have been changed now to activate whatever is in the selected slots (if possible). If the slot is selected, the outer border will be coloured. If the slot is selected, and the outfit can also be activated, the background will also be coloured.

This change was done so that if you change weapons, it won't affect your weapon slots.

**Comment by BM-W1** [comment]
Yes, when I click on it. With the weapons that's OK. The Photo-Voltaic Nanobot is a utility, however. I tried and clicked on all the icons (see screenshot). Is it intended this way?
Moreover, when I click on the Rackham's Razor item, it is added to the list of the chosen utilities on the main screen (see screenshot, too). What's the additional information or purpose for the player?
![Screenshot_20231022_184456](https://github.com/naev/naev/assets/122737685/1688e26a-efb9-4843-8e3e-e8ebf9f99050)

**Comment by bobbens** [comment]
OK, so Rackham's Razor should not appear there. That is a bug, I should have fixed it, thanks!

Being able to colour everything is normal now. The motivation is that, with the old system, if you are not using automatic weapon sets. If you swap out your primary weapons, say the turrets you have for some other turrets, the slots would become unselected and you would have to select them again in the info->weapons screen. Now, since the slots stay selected, you would be able to work with them right away.

Maybe the issue is that it's not clear to the player? I did the easy solution of just colouring the border, but maybe some other way of indicating selected but not active would be better?

**Comment by bobbens** [comment]
I guess the difference is that before, you always have to modify weapon sets after changing outfits.
Now, if you change for equivalent outfits, it's not necessary. So if you swap out forward weapon for forward weapon, it should still be usable. However, if you swap a forward weapon for a seeker or fighter bay, you probably have to change again. So it's nicer if you change stuff for stuff that's functionally equivalent, but if not, you have to change weapon sets again.
